### PR TITLE
micronaut-data issue 1378

### PR DESCRIPTION
**Pull request by rawilder**
With regards to this issue: https://github.com/micronaut-projects/micronaut-data/issues/1378

In cases where a field matches a getter exactly, a property should not be extracted but rather just passed through.

kotlin example:
```kotlin
class TestEntity(val isDeleted: Boolean)
```
turns into jvm byte code roughly representing this java code:
```java
class TestEntity {
    private Boolean isDeleted;
    public Boolean isDeleted() { return isDeleted; }
    // this part bums be out as it makes it seem difficult to correctly identify in the setter path
    // hence why I ignored it in this PR
    public void setDeleted(deleted: Boolean) {
        isDeleted = deleted;
    }
}
```

### CODE REVIEW

**Comment by graemerocher** [comment]
Needs a test

**Comment by rawilder** [comment]
test added

**Comment by dstepanov** [comment]
This doesn't look correct to me, I think the special case should be regarding `is` prefix for boolean fields. In your test case, the property name should be defined by its getter/setter and not by the private field name.

Please target the next major version, 3.4 which is going to be released soon so maybe 3.5.

**Comment by rawilder** [comment]
@dstepanov Property seems to be a pretty overloaded term so I had trouble actually nailing down what should be the value. The best I could tell in this context is that it's the combination of field and getter, so they should match in some capacity.

I'm fine with changing it to be a special case with `is` only, since the other test cases I wrote (getName backed by getName) seem like they really won't ever happen.


edit: also there is no 3.5.x branch. I'm happy to change it once available. For now I'll open it against 3.4.x

File: core/src/main/java/io/micronaut/core/naming/NameUtils.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
     private static final Pattern SERVICE_ID_REGEX = Pattern.compile("[\\p{javaLowerCase}\\d-]+");
     private static final String PREFIX_GET = "get";
     private static final String PREFIX_SET = "set";
-    private static final String PREFIX_IS = "is";
+    public static final String PREFIX_IS = "is";
```
Comment (rawilder) [inline core/src/main/java/io/micronaut/core/naming/NameUtils.java:41]: public may have been too aggressive here, I can put a static field in the inject lib instead if you deem this unnecessary

**Comment by dstepanov** [comment]
```
    public Boolean isDeleted() {
        return isDeleted;
    }

    public void setDeleted(Boolean isDeleted) {
        this.isDeleted = isDeleted;
    }
```
I would expect for this case the property name is `deleted` and not `isDeleted` in the same fashion as `getName/setName` results in `name` property.

**Comment by graemerocher** [comment]
yeah we need to be careful that this doesn't cause a regression in the Java side because of Kotlin oddities

**Comment by rawilder** [comment]
I mean this is definitely a kotlin oddity which is partially so annoying. Interested to hear y'all's thoughts. My understanding of property is it's an ambiguous term that generally refers to a combination of class field, getter, and setter. A lot of the use cases I have seen seem to ignore the setter entirely and just deduce "property" from getters strictly. 

How do y'all think we should approach this. I would consider this a bug, a property is being misidentified in the context of other parts of the framework. Namely micronaut data and openapi.

Data is producing bad HQL statements and openapi is producing incorrect REST documentation which leads to 400 bad requests when a consumer reads the swagger docs.

**Comment by dstepanov** [comment]
I have an alternative PR https://github.com/micronaut-projects/micronaut-core/pull/7094

**Pull request closed without merging**